### PR TITLE
code hygiene: using host functions instead of constructing manually

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

N/A

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by damiannolan — APPROVED** [review]
Nice cleanup!

File: modules/core/02-client/keeper/keeper.go [hunk 1]
```diff
@@ -389,6 +388,6 @@ func (k Keeper) GetAllClients(ctx sdk.Context) []exported.ClientState {
 // ClientStore returns isolated prefix store for each client so they can read/write in separate
 // namespace without being able to read/write other client's data
 func (k Keeper) ClientStore(ctx sdk.Context, clientID string) sdk.KVStore {
-	clientPrefix := []byte(fmt.Sprintf("%s/%s/", host.KeyClientStorePrefix, clientID))
+	clientPrefix := host.PrefixedClientStoreKey([]byte(clientID))
```
Comment (damiannolan) [inline modules/core/02-client/keeper/keeper.go:391]: seems like these are not quite the same, there's a trailing slash here but not in `host.PrefixedClientStoreKey()`

Comment (crodriguezvega) [reply modules/core/02-client/keeper/keeper.go]: Ahhh, yeah! I will revert then, but I wonder then why that trailing slash is needed there.. 🤔

Comment (damiannolan) [reply modules/core/02-client/keeper/keeper.go]: I'm not 100% as I didn't look super deep into it. I tried adding the trailing slash to [PrefixedClientStorePath](https://github.com/cosmos/ibc-go/blob/main/modules/core/24-host/keys.go#L48) but had failing tests elsewhere.

**Review by colin-axner — APPROVED** [review]
ACK, I think we can remove the backport label since this is code-hygiene?

**Pull request merged**